packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -116,8 +117,27 @@ const VeniaAdapter = props => {
  * other modules in the codebase need access to it.
  */
 VeniaAdapter.apolloLink = apiBase => {
+    // Intercept and shrink URLs from GET queries. Using
+    // GET makes it possible to use edge caching in Magento
+    // Cloud, but risks exceeding URL limits with default usage
+    // of Apollo's http link. `shrinkGETQuery` encodes the URL
+    // in a more efficient way.
+    const customFetchToShrinkQuery = (uri, options) => {
+        let url = uri;
+        if (options.method === 'GET') {
+            url = shrinkGETQuery(uri);
+        }
+        return fetch(url, options);
+    };
+
     return createHttpLink({
-        uri: apiBase
+        uri: apiBase,
+        fetch: customFetchToShrinkQuery,
```

Comment:
I just could not find a better way to hook into Apollo. Happy to hear other suggestions, but hard to modify the behavior of a terminating link, from what I saw.

Reply:
I think this is fine - the docs suggest that you do this if you need to target browsers where `fetch` is not natively provided. This seems to be the right place to do this.